### PR TITLE
feat: refactor error handling and add tests

### PR DESCRIPTION
**Pull request by gfyrag**
Fixes ENG-9

### CODE REVIEW

**Review by flemzord — APPROVED** [review]

File: components/ledger/internal/api/v1/controllers_accounts.go [hunk 1]
```diff
@@ -7,19 +7,17 @@ import (
 	"strconv"
 	"strings"
 
-	"github.com/formancehq/ledger/internal/api/shared"
+	"github.com/formancehq/ledger/internal/api/backend"
```
Comment (paul-nicolas) [inline components/ledger/internal/api/v1/controllers_accounts.go:10]: I have a global question here: should we add a lint in order to group imports ? (just my own opinion on it, feel free to tell me what you think)

File: components/ledger/internal/api/v1/controllers_accounts.go [hunk 2]
```diff
@@ -133,48 +130,46 @@ func getAccount(w http.ResponseWriter, r *http.Request) {
 
 	acc, err := l.GetAccountWithVolumes(r.Context(), query)
 	if err != nil {
-		ResponseError(w, r, err)
+		sharedapi.InternalServerError(w, r, err)
```
Comment (paul-nicolas) [inline components/ledger/internal/api/v1/controllers_accounts.go:133]: If the account does not exist, should we send a NotFound error instead?

File: components/ledger/internal/api/v2/controllers_accounts.go [hunk 3]
```diff
@@ -80,55 +86,53 @@ func getAccount(w http.ResponseWriter, r *http.Request) {
...
-		sharedapi.BadRequest(w, shared.ErrValidation, err)
+		sharedapi.BadRequest(w, ErrValidation, err)
 		return
 	}
 	query.PITFilter = *pitFilter
 
 	acc, err := l.GetAccountWithVolumes(r.Context(), query)
 	if err != nil {
-		shared.ResponseError(w, r, err)
+		sharedapi.InternalServerError(w, r, err)
```
Comment (paul-nicolas) [inline components/ledger/internal/api/v2/controllers_accounts.go:96]: Same here s hould we send a not found error ?

File: components/ledger/internal/api/v1/controllers_accounts.go (hunk 2 above)
Comment (gfyrag) [reply components/ledger/internal/api/v1/controllers_accounts.go:133]: Account always exists ;)

File: components/ledger/internal/api/v2/controllers_accounts.go (hunk 3 above)
Comment (gfyrag) [reply components/ledger/internal/api/v2/controllers_accounts.go:96]: Same as above.

File: components/ledger/internal/api/v1/controllers_accounts.go (hunk 1 above)
Comment (gfyrag) [reply components/ledger/internal/api/v1/controllers_accounts.go:10]: We already use the goimports linter on the main .golangci.yml

**Review by paul-nicolas — APPROVED** [review]

**Pull request merged**